### PR TITLE
nixos/nginx/gitweb: add some (crucial) options

### PR DESCRIPTION
**Pull request by ksixty**
###### Motivation for this change
This replaces some hardcoded values in nginx's VirtualHosts's configuration with customizable options. Previous values are kept as default, so nothing should break for existing users.

Call it a minimal refactoring.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @gnidorah

### CODE REVIEW

File: nixos/modules/services/web-servers/nginx/gitweb.nix [hunk 1]
```diff
@@ -17,13 +18,45 @@ in
...
+      default = "nginx";
+      type = types.str;
+      description = ''
+        Group that the CGI process will belong to. (Set to <literal>config.services.gitolite.group</literal> if you are using gitolite.)
+      '';
+    };
+
+    virtualHost = mkOption {
+      default = "_";
+      type = types.string;
```
Comment (gnidorah) [inline nixos/modules/services/web-servers/nginx/gitweb.nix]: trace: warning: types.string is deprecated because it quietly concatenates strings

Comment (vanyaklimenko) [reply nixos/modules/services/web-servers/nginx/gitweb.nix]: My bad! Fixed that.

**Review by ghost — APPROVED** [review]
Works fine here, thanks! :)

File: nixos/modules/services/web-servers/nginx/gitweb.nix [hunk 2]
```diff
@@ -17,13 +18,45 @@ in
...
+      description = ''
+        User that the CGI process will belong to. (Default almost surely will do.)
+      '';
+    };
+
+    group = mkOption {
+      default = "nginx";
+      type = types.str;
+      description = ''
+        Group that the CGI process will belong to. (Set to <literal>config.services.gitolite.group</literal> if you are using gitolite.)
```
Comment (flokli) [inline nixos/modules/services/web-servers/nginx/gitweb.nix]: Why does it need to be set to gitolite if we're using gitolite? repos are not world-redable in the gitolite state directory?

File: nixos/modules/services/web-servers/nginx/gitweb.nix [hunk 3]
```diff
@@ -17,13 +18,45 @@ in
...
+      description = ''
+        Location to serve gitweb on.
+      '';
+    };
+
+    user = mkOption {
+      default = "nginx";
+      type = types.str;
+      description = ''
+        User that the CGI process will belong to. (Default almost surely will do.)
```
Comment (flokli) [inline nixos/modules/services/web-servers/nginx/gitweb.nix]:
```suggestion
        Existing user that the CGI process will belong to. (Default almost surely will do.)
```
This requires the user to already exist, so better make that clear.

File: nixos/modules/services/web-servers/nginx/gitweb.nix (hunk 2 above)
Comment (vanyaklimenko) [reply nixos/modules/services/web-servers/nginx/gitweb.nix]: > repos are not world-redable in the gitolite state directory?
They aren't.

**Comment by flokli** [comment]
LGTM. Could you squash the 3 commits together into 1?

Also, it'd be nice if you could add a NixOS test, so we could ensure things keep working ;-)

**Comment by aanderse** [comment]
ping @vanyaklimenko

**Comment by vanyaklimenko** [comment]
@flokli @aanderse 

Sorry for the long wait. I've squashed the commits.

Not sure if the tests are really needed as the package seems pretty straightforward to me, and, alas, at the moment I just do not have enough time to spare on this. Let's leave it as it is for now?

**Comment by aanderse** [comment]
@vanyaklimenko NixOS tests are always nice for modules as at bare minimum it gives committers who have no domain specific knowledge of a module a better indication they aren't breaking anything by hitting 'merge'.

That being said, this module doesn't currently have a test and as your PR mentions these are crucial options. If anyone has a chance to come back to this and create a NixOS test full bonus points will be awarded :tada:

Merging based on the code seeming reasonable to myself and @flokli as well as testing provided by @gnidorah. Thanks for your contribution!

**Pull request merged**